tokio-util/src/task/spawn_abortable.rs

```diff
@@ -0,0 +1,100 @@
+//! An [`AbortOnDropHandle`] is like a [`JoinHandle`], except that it
+//! will abort the task as soon as it is dropped.
+
+use tokio::task::{AbortHandle, JoinError, JoinHandle};
+
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+/// A wrapper around a [`tokio::task::JoinHandle`],
+/// which [aborts] the task when it is dropped.
+///
+/// [aborts]: tokio::task::JoinHandle::abort
+#[must_use = "Dropping the handle aborts the task immediately"]
+#[derive(Debug)]
+pub struct AbortOnDropHandle<T>(JoinHandle<T>);
+
+impl<T> Drop for AbortOnDropHandle<T> {
+    fn drop(&mut self) {
+        self.0.abort()
+    }
+}
+
+impl<T> AbortOnDropHandle<T> {
+    /// Create an [`AbortOnDropHandle`] from a [`JoinHandle`].
+    pub fn new(handle: JoinHandle<T>) -> Self {
+        Self(handle)
+    }
+
+    /// Spawn a task and return an [`AbortOnDropHandle`] associated with it.
+    pub fn spawn(future: impl Future<Output = T> + Send + 'static) -> Self
+    where
+        T: Send + 'static,
+    {
+        Self(tokio::spawn(future))
+    }
```

Comment:
The spawn methods are usually standalone. I'm thinking that a standalone method makes more sense here too.

Reply:
Could `AbortOnDropHandle::new` or rather `AbortOnDropHandle::from_join_handle` be the only entry point? Otherwise, I'm not sure what to call the standalone function, `spawn_abort_on_drop`?